### PR TITLE
Fix template samples

### PR DESCRIPTION
**Pull request by Ivanidzo4ka**
## Description
Mixing bool and string force template engine to throw error.
So switch to 'auto' / 'tool' string values

## Related issues
Addresses [#1660].

## Testing
Manually

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Skip (it's a change in samples)

### CODE REVIEW

**Comment by LTA-Thinking** [comment]
The issue here is that we have two conflicting requirements for this parameter:
* When Cosmos is the datastore the user shouldn't have to put anything in the field.
* When SQL is the datastore there shouldn't be a default value so the user has to manually pick if they want automatic updates or not.

This was why I had initially tried to make it so the default value was false when using Cosmos, or not set if using SQL.
@rbans96, what is your thought?

**Comment by LTA-Thinking** [comment]
Relevant PR: #1301

File: samples/templates/default-azuredeploy-docker.json [hunk 1]
```diff
@@ -146,7 +146,7 @@
         },
         "sqlSchemaAutomaticUpdatesEnabled": {
             "type": "bool",
-            "defaultValue": "[if(equals(parameters('solutionType'),'FhirServerCosmosDB'), bool('false'), '')]",
+            "defaultValue": "[if(equals(parameters('solutionType'),'FhirServerCosmosDB'), false(), true())]",
```
Comment (brendankowitz) [inline samples/templates/default-azuredeploy-docker.json:149]: I think the intended behavior was to have no default value when sql to force a selection.

- The above is only true when the datalayer targets SQL.
- The value `true` is know to have problems when running on a webfarm (should be dev/test only for now)

**Comment by Ivanidzo4ka** [comment]
How about now?

Pass validation if it's cosmos, and complains about incorrect value if it's sql. (I wish I could copy error message, but for some reason azure localize error string for me, so I don't think it would be useful for English speaking people)


---
In reply to: [786959225](https://github.com/microsoft/fhir-server/pull/1700#issuecomment-786959225) [](ancestors = 786959225)

**Comment by Ivanidzo4ka** [comment]
From what I see it's either this, and I need to change yml file for pipelines (to change it from boolean to string) which is somewhat ugly from developer point of view, but I can't find any other way to have unset boolean value, or we can set default value to false for all cases.

---
In reply to: [786974696](https://github.com/microsoft/fhir-server/pull/1700#issuecomment-786974696) [](ancestors = 786974696,786959225)

**Review by LTA-Thinking — APPROVED** [review]
This looks like the best option we have to me, but I'd like @brendankowitz opinion as well.

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**